Python/heap.py

```diff
@@ -0,0 +1,71 @@
+# Implementation of heap from list in Python
+
+class Heap(object):
+
+    def __init__(self):
+        self.list = []
+        self.last_index = -1
+
+    def insert(self, x):
+        self.last_index += 1
+        if not self.list:
+            self.list.append(x)
+            return
+        self.list.append(x)
+        index = self.last_index
+        parent = int((index-1)/2)
+        while (self.list[index] > self.list[parent]) and parent >= 0:
+            temp = self.list[index]
+            self.list[index] = self.list[parent]
+            self.list[parent] = temp
+            index = parent
+            parent = int((parent-1)/2)
+
+    def max(self):
+        return self.list[0]
+
+    def pop(self):
+        head = self.list[0]
+        self.list[0] = self.list[self.last_index]
+        self.list.pop()
+        print self.list
+        self.last_index -= 1
+        index = 0
+        left_child_index = (index * 2) + 1
+        right_child_index = (index * 2) + 2
+        if left_child_index > self.last_index: # 0 or 1 nodes in heap, done
+            return head
+
+        if right_child_index > self.last_index: # no right child
+            right_child_index = index
+
+        while (self.list[index] < self.list[right_child_index]) or \
```

Comment:
20 line while loop prolly indicates you want a helper function. Also needs commenting. What is going on inside this loop? No one wants to read a loop this long.